ff/src/fields/mod.rs

```diff
@@ -350,6 +350,8 @@ pub trait Field:
         }
         Some(res)
     }
+
+    fn mul_by_base_field_elem(&self, elem: &Self::BasePrimeField) -> Self;
```

Comment:
```suggestion
    fn mul_by_base_prime_field(&self, elem: &Self::BasePrimeField) -> Self;
```

Also, should we move this into a `Mul<Self::BasePrimeField, Output = Self>` bound? (And maybe add `for<'a> Mul<&'a Self::BasePrimeField, Output = Self>`, `MulAssign<Self::BasePrimeField>`, and `for<'a> MulAssign<&'a Self::BasePrimeField>` bounds too?

Reply:
I tried this initially, but we already have `Mul<Self>` coming from a [macro](https://github.com/arkworks-rs/algebra/blob/3835af548d929d08f3ff1771c4349ed49c698efd/ff/src/fields/arithmetic.rs#L171), and adding a trait bound as you suggest, and then implementing:
```

impl<P: QuadExtConfig> Mul<<Self as Field>::BasePrimeField> for QuadExtField<P> {
    type Output = Self;
    ...
}
```
causes a `conflicting implementation` error.
I believe this is what you meant, right?

Reply:
Ouch that's annoying...

I guess Rust doesn't know that for `QuadExtField`, `BasePrimeField` is always different from `Self`...

But it should be able to figure that out: `BasePrimeField: PrimeField`, while `Self` doesn't implement that.